libs/ledger-live-common/src/families/celo/preload.ts

```diff
@@ -0,0 +1,79 @@
+import { BigNumber } from "bignumber.js";
+import { Observable, Subject } from "rxjs";
+import { log } from "@ledgerhq/logs";
+import { PRELOAD_MAX_AGE } from "./logic";
+import type { CeloPreloadData, CeloValidatorGroup } from "./types";
+import { getValidatorGroups } from "./validators";
+
+let currentCeloPreloadedData: CeloPreloadData = {
+  validatorGroups: [],
+};
+
+function fromHydrateValidator(
+  validatorGroupRaw: Record<string, any>
+): CeloValidatorGroup {
+  return {
+    address: validatorGroupRaw.address,
+    name: validatorGroupRaw.name,
+    votes: new BigNumber(validatorGroupRaw.votes),
+  };
+}
+
+function fromHydratePreloadData(data: any): CeloPreloadData {
+  let validatorGroups = [];
+
+  if (typeof data === "object" && data) {
+    if (Array.isArray(data.validatorGroups)) {
+      validatorGroups = data.validatorGroups.map(fromHydrateValidator);
+    }
+  }
+
+  return {
+    validatorGroups,
+  };
+}
+
+const updates = new Subject<CeloPreloadData>();
+export function getCurrentCeloPreloadData(): CeloPreloadData {
+  return currentCeloPreloadedData;
+}
+export function setCeloPreloadData(data: CeloPreloadData) {
```

Comment:
I think we can omit the `export` here since this function is only referenced in this file.